lib/macho/macho_file.rb

```diff
@@ -233,10 +264,12 @@ def dylib_id=(new_id)
       raise ArgumentError.new("new ID must be a String") unless new_id.is_a?(String)
       return unless dylib?
 
-      dylib_cmd = command(:LC_ID_DYLIB).first
-      old_id = dylib_id
+      old_cmd = command(:LC_ID_DYLIB).first
+      new_cmd = LoadCommand.create(:LC_ID_DYLIB, new_id, old_cmd.timestamp,
+        old_cmd.current_version, old_cmd.compatibility_version)
 
-      set_name_in_dylib(dylib_cmd, old_id, new_id)
+      add_command(new_cmd)
+      delete_command(old_cmd)
```

Comment:
I applaud the simplicity created by this, but I think we might want to additionally implement a `replace_command` method to avoid reshuffling commands just because they are being modified.

This shouldn't be an issue here, but it can definitely be problematic for the family of `LC_LOAD_DYLIB` commands where load order can also influence against which dylib a symbol is resolved and also for `LC_RPATH` commands as this can affect what library ends up being loaded.

Another problem in this specific case is that this can fail even if there is enough space for the changed dylib ID, simply because the new one is added before the old one is removed, thus temporarily there needs to be space for both load commands, which is more than would be strictly necessary.